src/memray/_memray/native_resolver.cpp

```diff
@@ -463,4 +463,46 @@ SymbolResolver::currentSegmentGeneration() const
     return d_segments.size();
 }
 
+std::vector<std::string>
+unwindHere()
+{
+    struct CallbackData
+    {
+        std::vector<std::string> frames;
+        struct backtrace_state* state;
+    };
+
+    auto err_callback = [](void* data, const char* msg, int errnum) { return; };
+
+    auto callback = [](void* vdata, uintptr_t pc, const char* filename, int lineno, const char* function)
+            -> int {
+        auto result = reinterpret_cast<CallbackData*>(vdata);
+        std::string the_function = function ? function : "<unknown>";
+        std::string the_filename = filename ? filename : "<unknown>";
+        if (!function && !filename) {
+            // Fallback callbacks for when we can't get a filename or function name via debug
+            // information. These fallback callbacks query the symbol table instead.
+            auto fallback_callback =
+                    [](void* data, uintptr_t, const char* symbol, uintptr_t, uintptr_t) {
+                        auto result = reinterpret_cast<std::vector<std::string>*>(data);
+                        std::string the_function = symbol ? symbol : "<unknown>";
+                        result->push_back(the_function + ":<unknown>:<unknown>");
```

Comment:
You're doing `"<unknown>"` for the line number here - perhaps it should be `0` instead, so the caller doesn't need to worry about whether it's a string or an integer...

But perhaps it doesn't matter much, since we're not intending to use this in more places.

Reply:
I went with empty string